### DISCUSSION TITLE
A proplem with ImageRegion

### DISCUSSION THREAD

**Discussion by modean97** [Spawnova/ShinsImageScanClass#31] [General]
Thanks for your effort and all works you've done k3.
I have a small issue with ImageRegion... 
This error appears when I convert the script into an exe, and run it.
Note: it only appears when i using ImageRegion.
Is there any solution or suggestion?
![image](https://github.com/user-attachments/assets/2bb670dd-4761-4427-a82e-6d48f1e19cb4)

**Comment by Spawnova** [comment]
What versions of ahk are you using, and can you paste the line where you call ImageRegion? 
I'll have a look on my my end later today to see if I can replicate it

**Reply by modean97** [reply]
Hi,
I am using ahk 2.0.12

this is an example code

`#Requires AutoHotkey v2.0

#Include ../ShinsImageScanClass/AHKv2/ShinsImageScanClass.ahk

scan := ShinsImageScanClass()

f1 up::
{
    if (scan.ImageRegion("test.bmp", 0, 0, 500, 500, 0, &FoundX, &FoundY))
    {
        MouseMove(FoundX, FoundY)
    }
}`


which works perfectly as .ahk file or from the vscode debugger 
but after being compiled, 
when i press f1 this message shows up
![image](https://github.com/user-attachments/assets/34e60d63-400c-492e-817d-72e4cf61a105)

thanks for your time and quick response.

**Reply by Spawnova** [reply]
Ok awesome, I'll look into it and update you tomorrow, thanks for reporting the problem

**Reply by Spawnova** [reply]
Okay, I had some free time tonight actually, it seems like it may be an issue with the 32 bit compiled machine code, I will try and fix it tomorrow, using ahk in 64bit mode should fix it for now though, and I do recommend using 64bit as the default version anyway if it's possible

**Reply by modean97** [reply]
Ahaa
Never imagined compiling with 32 would causes that.
I'll try asap.
Thanks alot for this masterpiece class and best wishes for you.

**Reply by Spawnova** [reply]
It's a bit more complicated than I thought, due to being pretty busy irl it's going to take a little while to find the root issue unfortunately 
Hopefully 64bit version is working for you in the meantime

I do have a new version I'm working on as well that will instead make use of a custom dll, which will solve this issue and open up more advanced techniques for getting pixel data on applications that don't support BitBlt =P

I'll let you know though once I've figured this problem out

**Reply by modean97** [reply]
Thank you for the update! 
I understand that you're busy and appreciate your effort in resolving the issue. 
The 64-bit version is working fine for now. 
Looking forward to trying out the new version with the custom DLL when it's ready.
Thanks again for keeping me posted!
